hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -3,18 +3,1146 @@
+
+  // Replicate CSR and mem tests over TL and JTAG interfaces.
+  intf: ["_tl", "_jtag"]
+
+  // TODO: remove the common testplans if not applicable
   import_testplans: ["hw/dv/tools/testplans/csr_testplan.hjson",
+                     "hw/dv/tools/testplans/enable_reg_testplan.hjson",
                      "hw/dv/tools/testplans/shadow_reg_errors_testplan.hjson",
                      "hw/dv/tools/testplans/mem_testplan.hjson",
+                     "hw/dv/tools/testplans/intr_test_testplan.hjson",
                      "hw/dv/tools/testplans/tl_device_access_types_testplan.hjson",
                      "hw/ip/tlul/data/tlul_testplan.hjson"]
+
   entries: [
+    //////////////////////////////////////////////////////////////////////////////////
+    // IO Peripherals                                                               //
+    // UART, GPIO, I2C, SPIDEV, SPIHOST, USB, PINMUX & PADCTRL, RV_DM, PATTGEN, PWM //
+    //////////////////////////////////////////////////////////////////////////////////
+
+    // UART (pre-verified IP) integration tests:
     {
-      name: sanity
-      desc: ''' Basic hello world type test.
+      name: chip_uart_tx_rx
+      desc: '''Verify transmission of data over the TX and RX port.
+
+            SW test sends a known payload over the TX port. The testbench, at the same time
+            sends a known payload over RX. On reception, both payloads are checked for integrity.
+            SW validates the reception of TX watermark, RX watermark, and the TX empty interrupts.
+            Verify each UART instance at the chip level independently.
             '''
       milestone: V1
-      tests: ["chip_sanity"]
+      tests: ["chip_uart_tx_rx"]
+    }
+    {
+      name: chip_uart_rx_overflow
+      desc: '''Verify the RX overflow interrupt.
+
+            The testbench sends a random payload of size greater than the RX fifo size (32). The SW
+            ignores the received the data to allow the RX overflow interrupt to assert.
+            Verify each UART instance at the chip level independently.
+            '''
+      milestone: V1
+      tests: ["chip_uart_tx_rx"]
+    }
+
+    // GPIO (pre-verified IP) integration tests:
+    {
+      name: chip_gpio_out
+      desc: '''Verify GPIO outputs.
+
+            SW test configures the GPIO to be in the output mode. The test walks a 1 through the
+            pins. The testbench checks the value for correctness.
+            '''
+      milestone: V1
+      tests: []
+    }
+    {
+      name: chip_gpio_in
+      desc: '''Verify GPIO inputs.
+
+            The SW test configures the GPIO to be in input mode and enables all of them to generate
+            an interrupt. The testbench walks a 1 through the pins. SW test ensures that the
+            interrupt corresponding to the right pin is seen.
+            '''
+      milestone: V1
+      tests: []
+    }
+
+    // SPI_DEVICE (pre-verified IP) integration tests:
+    {
+      name: chip_spi_device_tx_rx
```

Comment:
if we end up doing passthrough for this iteration, this could get significantly more complicated.
But probably no need to worry about it for now..until we get a better idea what features SPI needs to have. 

Do you think it makes sense to have a max-frequency type test also? This won't mean much for RTL, but we could ask Nuvoton to run it in GLS